core/core.go

```diff
@@ -794,6 +789,10 @@ func (n *IpfsNode) LoadPrivateKey() error {
 		return err
 	}
 
+	if cfg.Identity.PrivKey == "" {
+		return nil
+	}
+
```

Comment:
Might cause panics in some places, but that should only really affect tests

Reply:
I believe I've caught most places. Basically, everywhere we need the private key and would have otherwise called `LoadPrivateKey` or `SetupOfflineRouting`, we need to check if this is nil.